v7.10.0 (#13806)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.10.0

_Jul 11, 2024_

We'd like to offer a big thanks to the 12 contributors who made this release possible. Here are some highlights ✨:

- 🎁 Add selectors to support showing child row count in footer in the Data Grid

> - 🧰 Add utility function to check if an optional plugin is present in the Tree View

Reviewer: I'm not sure we need to highlight this mainly internal addition..? 🤔 
cc @flaviendelangle @noraleonte

Reviewer 2: Hmm, I don't think so. It's an internal thing 🤔

Reviewer: ```suggestion
```

- 🌍 Improve Spanish (es-ES) locale on the Data Grid
- 🐞 Bugfixes
- 📚 Documentation improvements

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

### Data Grid

#### `@mui/x-data-grid@7.10.0`

- [DataGrid] Add selectors to support showing child row count in footer (#13725) @KenanYusuf
- [DataGrid] Fix incorrect panels position when using a toolbar (#13474) @oukunan
- [DataGrid] Use `readonly` on more array props (#13331) @pcorpet
- [DataGrid] set default variant to` 'standard'` in `GridFilterInputMultipleValue` (#13129) @tarunrajput
- [l10n] Improve Spanish (es-ES) locale (#13772) @joserealdev

#### `@mui/x-data-grid-pro@7.10.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.10.0`, plus:

- [DataGridPro] Keep bottom pinned row at the bottom (#13313) @romgrk

#### `@mui/x-data-grid-premium@7.10.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.10.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@7.10.0`

> No updates since `@mui/x-date-pickers@7.9.0`

Author: I am not sure about this one. I have added this line manually, the section was generated empty.

Reviewer 3: Interesting, I think this is the first time this occurred. I like the way you structured it tho.

Reviewer: That's because there are no PRs that are directly related to only the MIT package.
Both merged fixes are for `pro`.
We could improve the changelog generation to always include the base package changelog entry if there are only paid package changes.
It might become more relevant once we start releasing `tree-view-pro` and `charts-pro`. 🤔

Reviewer: ```suggestion
- [fields] Prevent infinite recursion when ensuring selection (#13779) @LukasTy
- [fields] Unify fields behavior regardless of the `readOnly` flag (#13688) @LukasTy
```

#### `@mui/x-date-pickers-pro@7.10.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@7.10.0`, plus:

- [DateRangePicker] Fix `calendarHeader` slot props propagation (#13780) @LukasTy
- [DateTimeRangePicker] Resolve `format` from given `views` (#13743) @LukasTy

### Charts

#### `@mui/x-charts@7.10.0`

> - [charts] Fix area with log scale (#13791) @alexfauquette

Reviewer 4: ```suggestion
- [charts] Fix displaying area of a `LineChart` when using the `log` scale (#13791) @alexfauquette
```
